test/support/utils_invoke.h

```diff
@@ -219,6 +219,16 @@ struct invoke_on_all_hetero_policies
             auto my_policy = make_new_policy<kernel_name>(queue);
             iterator_invoker<::std::random_access_iterator_tag, /*IsReverse*/ ::std::false_type>()(
                 my_policy, op, ::std::forward<Args>(rest)...);
+
+#if !__SYCL_UNNAMED_LAMBDA__
+            if (false)
```

Comment:
Some run-time `false`-condition check just to compile Kernel's code.

Reply:
Yeah, this is the question I have as well.  One way I proposed we could do this is to read a value from a file at runtime which we know to be "false", but the compiler does not. 

The reason I prefer something like this where we can guarantee it is a real thing only knowable at runtime, but also deterministically `false`, is that dead code elimination may depend on compiler internals, and may change over time, so even if it works now it may not work in the future.

Reply:
I think something like this would work: https://godbolt.org/z/9n7GW7s7q

Just define some variable `volatile bool always_false = false;` Reads to the volatile variable cannot be optimized out, so I think the contents within the if statement will always be compiled.

Reply:
Thats a good idea.

Reply:
Fixed.